fishing.dm

```diff
@@ -0,0 +1,287 @@
+//			fishcatch()
+//		if(prob(100-baited)) //100 minus the probability gives us the remaining probability that it fucks up
+////			user.visible_message("<span class='warning'>Your line dangles there akwardly as no fish take the bait.</span>")
+//			return
+//
+//		return ..()
+//
+	if(istype(I, /obj/item/weapon/fishing/pulser))
+		user.visible_message("<span class='warning'>[user] places a pulser on top of the [src]</span>")
+//
+	if(!istype(I, /obj/item/weapon/fishing/pulser))
+		user.visible_message("<span class='warning'>[user] dips something into the [src]</span>")
+		return ..()
+
+//I will handle all the fishy calculations in this attackby, so it draws carp up and rolls the dice etc. etc.
+
+
+//This will be phased out once i get the code actually working, this is just for prototyping
+
+
+
+
+
+
+
+
+///Items/////////////////////////////////////////////
+//												   //
+/////////////////////////////////////////////////////
+
+
+
+//EXPERIMENTAL//
+
+
+
+
+/////////Bait//////////////////
+
+
+
+
+/obj/item/weapon/fishing/pole
+	name = "Fishing Pole"
+	desc = "A basic rod made from wire and metal"
+	icon = 'icons/obj/fishing.dmi'
+	icon_state = "pole" //remember to change this
+	var/chance = 30 //lowest chance for getting a fisherino, playtest to see
+	var/bait_power = 0 //when bait added, chance increased will put in an if(then chance * bait power etc.)
+	var/cancatch = 1 //What carp types can we catch? higher numbers = more rare carp
+	var/baited
+	var/carp_type = /mob/living/simple_animal/hostile/carp
+	var/carpnumber = 1
+	var/fishingspeed = 50 //5 seconds to catch a fish potentially
+
+/obj/item/weapon/fishing/pole/admin
+	name = "ADMIN Fishing Pole"
+	desc = "REEEEEE"
+	chance = 100
+	cancatch = 8 //god pole catches all, and never uses bait
+	fishingspeed = 0 //GOD POLE
+
+
+/obj/item/weapon/fishing/pole/attackby(obj/item/I, mob/user, params)	//baiting uk shit pls kick
+	if(istype(I, /obj/item/weapon/fishing/bait))
+		var/obj/item/weapon/fishing/bait/B = I
+		bait_power = B.bait_power
+		cancatch = B.cancatch
+		icon_state = "pole_bait"
+		user.visible_message("<span class='warning'>[user] adds the [I] to the [src] creating an effective lure.</span>")
+		qdel(B)
+
+	if(istype(I, /obj/item/stack))
+		var/obj/item/stack/B = I
+		//B.Split(1,2)
+		bait_power = B.bait_power
+		cancatch = B.cancatch
+		icon_state = "pole_bait"
+		user.visible_message("<span class='warning'>[user] adds the [I] to the [src] creating a crude lure.</span>")		//dump minerals on rods for lures
+		qdel(B) //will likely kill popped stack as well, need to test
+
+	else
+		user.visible_message("<span class='warning'>[user] slaps their [src] with the [I].</span>")
+
+
+
+/obj/item/weapon/fishing/pole/proc/rollthedice() //WHAT CARP ARE WE GONNA GET OH BOY OH BOY
+	var/list/carppossibilities = list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/hostile/carp/megacarp)		//all the carps it can be, we will change this with cancatch
```

Comment:
loop through subtypes of carp instead of manualy making a list. You can then have a rarity or whatever var for determining if it's in the list based on cancatch